### ISSUE TITLE
"404 - URL not provided in the app Scully is serving" when calling backend with a proxy

### ISSUE THREAD

**Issue by ufoscout** [scullyio/scully#1298]
### Description
I am trying to prerender some pages that perform an HTTP call to the backend; so, I configured a proxy to forward all `/api/*` calls to a running backend. The issue is that when I launch `npm run scully`, every call to `/api` returns a "404 - URL not provided in the app Scully is serving" error.

#### 🔬 Minimal Reproduction
My scully configuration in package.json:
```json
{
  "name": "my_project",
  "version": "0.0.0",
  "scripts": {
    "start": "ng serve --hmr --proxy-config proxy.conf.json",
    "scully": "npx scully --proxy-config proxy.conf.json -- --project my_project",
    "scully:serve": "npx scully serve --proxy-config proxy.conf.json -- --project my_project"
  },

  "dependencies": {
    "@angular/cdk": "^11.2.5",
    "@angular/common": "~11.2.5",
    "@angular/compiler": "~11.2.5",
    "@angular/core": "~11.2.5",
    ...
    "@scullyio/init": "^1.1.1",
    "@scullyio/ng-lib": "^1.0.0",
    "@scullyio/scully": "^1.0.0",
    ...
  },
  ...
}

```

The proxy file:
```json
{
  "/api": {
    "target": "https://app_url.com/",
    "secure": false,
    "changeOrigin": true
  }
}
```

Please note that the very same proxy file is used by `npm start` and it works as expected.

I also tried rewriting this from:
`"scully": "npx scully --proxy-config proxy.conf.json -- --project my_project",`
to 
`"scully": "npx scully -- --project my_project --proxy-config proxy.conf.json",`
but the result doesn't change.

#### 💻Your Environment

**Angular Version:**

<pre><code>
Angular CLI: 11.2.5
Node: 14.15.0
OS: linux x64

Angular: 11.2.6
... animations, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1102.5
@angular-devkit/build-angular      0.1102.5
@angular-devkit/core               11.2.5
@angular-devkit/schematics         11.2.5
@angular/cdk                       11.2.5
@angular/cli                       11.2.5
@angular/material                  11.2.5
@angular/material-moment-adapter   11.2.5
@schematics/angular                11.2.5
@schematics/update                 0.1102.5
rxjs                               6.6.6
typescript                         4.1.5

</code></pre>

**Scully Version:**

<pre><code>
@scullyio/init ^1.1.1
@scullyio/ng-lib  ^1.0.0
@scullyio/scully  ^1.0.0

</code></pre>

## 🔥 Exception or Error

<pre><code>
404 - URL not provided in the app Scully is serving
The url "/api/my_rest_endpoint" is not provided in the scully.routes.json, so it can't be generated
</code></pre>

**Comment by ufoscout** [comment]
Hi @SanderElias, as you marked this as a question, I suppose there is an answer too :)
We are blocked on this issue, any advice?

**Comment by aaronfrost** [comment]
@ufoscout any chance you have a repo we can see? Or can we set up a call?

**Comment by ufoscout** [comment]
@aaronfrost I cannot share it, it's not a public project. Let's have a call if you can, let me know your availabilities.

**Comment by SanderElias** [comment]
@ufoscout Sorry. I meant to answer it right away but got pulled away.
Have a look [at this ](https://scully.io/docs/Reference/config/#handle404). 
We are using the proxy in our testbed, so if it would be this broken, I would know. 
In our testbed we have this in the config file:
```typescript
    handle404: 'baseOnly',
    proxyConfig: 'proxy.conf.js',
```
What you are seeing is the default behavior.  We set our defaults for the majority of use-cases.

**Comment by ufoscout** [comment]
Hi @SanderElias ,
if that is the default behaviour, how can it be customized?

I think that the need for calling a backend is a pretty common scenario for Angular developers; supporting this use case is fundamental for Scully to become more and more popular. Otherwise, it will be usable only for static cases where other frameworks are much more affirmed than Angular.

@aaronfrost, I created a small reproducer [here](https://github.com/ufoscout/angular-scully-issue) along with the steps to use it. Let me know if it's enough.

**Comment by SanderElias** [comment]
@ufoscout By setting the handle404 setting in your config. As documented in the link I provided you with.

**Comment by ufoscout** [comment]
@SanderElias 
About the link you provided, thanks for it, but I have to say that that documentation is a little bit too thin to be of any help.
Anyway, I solved the issue by adding this to the `scully.my_app.config.ts` file:
```ts
export const config: ScullyConfig = {
  ...
  proxyConfig: 'proxy.conf.json',
  ...
}
```

Instead, if I add these two entries as suggested by you:
```ts
    handle404: 'baseOnly',
    proxyConfig: 'proxy.conf.js',
```
then `npm run scully` returns 404 error on every route.

**Comment by SanderElias** [comment]
For your setup you need:
```typescript
handle404: 'index'
```

**Comment by SanderElias** [comment]
Closing as this discussion seems resolved. 
If there is still actionable content here, please let us know, and we will reopen or create a follow-up issue
